Unused parameter warnings for pmix_util_getid in pmix_getid.c

<!--Thank you for taking the time to submit an issue!-->

## Background information
I am trying to compile a cloned version of OpenMPI including OpenPMIx under Solaris 11.4 where I get compilation errors.
### What version of the PMIx Reference Library are you using? <!--(e.g., v1.0, v2.1, git master @ hash, etc.)-->
Commit https://github.com/openpmix/openpmix/commit/5bd97056ccd994c711fde4ac19aded8e4eac8285
### Describe how PMIx was installed <!--(e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)-->
OpenMPI cloned from current release, i.e. commit dd6b87590cd767f5ff2047ce117a4f16a216b628, followed by

```sh
export FC=/opt/ulm/dublin/cmd/gfortran
export CC=/opt/ulm/dublin/cmd/gcc
export CPPFLAGS=-D_XOPEN_SOURCE=700
export CXX=/opt/ulm/dublin/cmd/g++
export AUTOMAKE_JOBS=8

cd ompi-master &&
   perl autogen.pl &&
   bash ./configure --prefix=/opt/ulm/dublin \
      --disable-silent-rules \
      --libdir=/opt/ulm/dublin/lib/amd64 \
      --enable-wrapper-rpath &&
   make DESTDIR=/home/pkgdev/dublin/openmpi/proto install
```
### Please describe the system on which you are running

* Operating system/version: Solaris 11.4
* Compiler: gcc 11.2.0
* Computer hardware: 64 bit x86 architecture
* Network type: not relevant here

-----------------------------

## Details of the problem

Compilation ends up with following error:

```
libtool: compile:  /opt/ulm/dublin/cmd/gcc -D_XOPEN_SOURCE=700 -DHAVE_CONFIG_H -I. -I../../src/include -I../../include -iquote/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix -iquote/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/src -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/include -D_REENTRANT -O3 -DNDEBUG -D_REENTRANT -finline-functions -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wextra -Werror -MT pmix_getid.lo -MD -MP -MF .deps/pmix_getid.Tpo -c pmix_getid.c  -fPIC -DPIC -o .libs/pmix_getid.o
pmix_getid.c: In function 'pmix_util_getid':
pmix_getid.c:43:35: error: unused parameter 'sd' [-Werror=unused-parameter]
   43 | pmix_status_t pmix_util_getid(int sd, uid_t *uid, gid_t *gid)
      |                               ~~~~^~
pmix_getid.c:43:46: error: unused parameter 'uid' [-Werror=unused-parameter]
   43 | pmix_status_t pmix_util_getid(int sd, uid_t *uid, gid_t *gid)
      |                                       ~~~~~~~^~~
pmix_getid.c:43:58: error: unused parameter 'gid' [-Werror=unused-parameter]
   43 | pmix_status_t pmix_util_getid(int sd, uid_t *uid, gid_t *gid)
      |                                                   ~~~~~~~^~~
cc1: all warnings being treated as errors
```

The implementation of `pmix_util_getid` consists of a `#if`/`#elif`-chain:

```c
pmix_status_t pmix_util_getid(int sd, uid_t *uid, gid_t *gid)
{
#if defined(SO_PEERCRED)
[..]
#endif

#if defined(SO_PEERCRED) && (defined(HAVE_STRUCT_UCRED_UID) || defined(HAVE_STRUCT_UCRED_CR_UID))
[..]
#elif defined(HAVE_GETPEEREID)
[..]
#else
    return PMIX_ERR_NOT_SUPPORTED;
#endif

    return PMIX_SUCCESS;
}
```

Neither `SO_PEERCRED` nor `HAVE_GETPEEREID` are supported by Solaris 11.4. Hence, we will end up in the `#else` case with parameters _sd_, _uid_, and _gid_ being unused.

--- a/src/util/pmix_getid.c
+++ b/src/util/pmix_getid.c
@@ -11,7 +11,7 @@
  *                         All rights reserved.
  * Copyright (c) 2007      Cisco Systems, Inc.  All rights reserved.
  * Copyright (c) 2015-2020 Intel, Inc.  All rights reserved.
- * Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021-2023 Nanook Consulting.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -79,6 +79,7 @@ pmix_status_t pmix_util_getid(int sd, uid_t *uid, gid_t *gid)
         return PMIX_ERR_INVALID_CRED;
     }
 #else
+    PMIX_HIDE_UNUSED_PARAMS(sd, uid, gid);
     return PMIX_ERR_NOT_SUPPORTED;
 #endif
 

